### PR TITLE
Improve inbound testing and avoid auto-injected tls inspector on inbound

### PR DESCRIPTION
**Pull request by howardjohn**
Contains multiple commits:

*    Add transport protocol match to all inbound filter chains

    Fixes https://github.com/istio/istio/issues/27004

* Add test for sidecar inbound traffic

*     Improve simulation test to split TLS and protocol

    This will be useful for future changes

### CODE REVIEW

**Comment by howardjohn** [comment]
Hold to test behavioral change of matching filter chain and failing TLS vs not matching filter chain in strict mode

**Comment by howardjohn** [comment]
The only potential negative impact I can think of now is strict mTLS filter chain will now not match plaintext. This is actually a good thing I think - in combination with https://github.com/istio/istio/pull/27853 we will get access logs for this case.

We could also NOT add the tls_inspector for strict mode, but I do not see a benefit - all TLS traffic is NOT server first by definition, so no concerns there.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -195,27 +197,29 @@ var (
...
 			ApplicationProtocols: mtlsHTTPALPNs,
 			// If client sends mTLS traffic, transport protocol will be set by the TLS inspector
-			TransportProtocol: "tls",
+			TransportProtocol: xdsfilters.TLSTransportProtocol,
 			Protocol:          istionetworking.ListenerProtocolHTTP,
 		},
 		{
 			// client side traffic was detected as HTTP by the outbound listener, sent out as plain text
 			ApplicationProtocols: plaintextHTTPALPNs,
 			// No transport protocol match as this filter chain (+match) will be used for plain text connections
```
Comment (lambdai) [inline pilot/pkg/networking/core/v1alpha3/listener.go:206]: "use plain text match"
nit: comment could be fix later.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 2]
```diff
@@ -235,21 +239,25 @@ var (
 			// If we are in strict mode, we will get mTLS HTTP ALPNS only.
 			ApplicationProtocols: mtlsHTTPALPNs,
 			Protocol:             istionetworking.ListenerProtocolHTTP,
+			TransportProtocol:    xdsfilters.TLSTransportProtocol,
 		},
 		{
 			// Could not detect traffic on the client side. Server side has no mTLS.
```
Comment (lambdai) [inline pilot/pkg/networking/core/v1alpha3/listener.go:245]: Nit: legacy comment. It's under strict TLS here

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 3]
```diff
@@ -595,15 +602,24 @@ allChainsLabel:
...
 		var filterChainMatch *listener.FilterChainMatch
 
+		if chain.FilterChainMatch == nil {
+			chain.FilterChainMatch = &listener.FilterChainMatch{}
+		}
+		if chain.TLSContext == nil {
+			chain.FilterChainMatch.TransportProtocol = xdsfilters.RawBufferTransportProtocol
+		} else {
+			chain.FilterChainMatch.TransportProtocol = xdsfilters.TLSTransportProtocol
+		}
```
Comment (lambdai) [inline pilot/pkg/networking/core/v1alpha3/listener.go:612]: awesome!

File: pilot/pkg/networking/core/v1alpha3/gateway_simulation_test.go [hunk 4]
```diff
@@ -291,7 +292,7 @@ spec:
 					// TODO(https://github.com/istio/istio/issues/24638) This is a bug!
 					// We should not have multiple matches, envoy will NACK this
 					"call",
-					simulation.Call{Port: 443, Protocol: simulation.HTTPS, HostHeader: "foo.bar"},
+					simulation.Call{Port: 443, Protocol: simulation.HTTP, TLS: simulation.TLS, HostHeader: "foo.bar"},
```
Comment (lambdai) [inline pilot/pkg/networking/core/v1alpha3/gateway_simulation_test.go:295]: Thank you for the change!

**Review by lambdai — APPROVED** [review]
Thank you for this huge change!

**Review by ramaraochavali — APPROVED** [review]
LGTM

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**